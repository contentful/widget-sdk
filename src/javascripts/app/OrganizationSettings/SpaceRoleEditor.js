/* eslint "rulesdir/restrict-inline-styles": "warn" */
import React from 'react';
import PropTypes from 'prop-types';
import { without } from 'lodash';
import { css } from 'emotion';
import {
  Button,
  Checkbox,
  Dropdown,
  DropdownList,
  DropdownListItem
} from '@contentful/forma-36-react-components';
import tokens from '@contentful/forma-36-tokens';
import classNames from 'classnames';

import { SpaceRole as SpaceRoleProp } from 'app/OrganizationSettings/PropTypes';
import { ADMIN_ROLE, ADMIN_ROLE_ID } from 'access_control/constants';

const styles = {
  roleList: css({
    maxHeight: '300px',
    overflowY: 'auto'
  }),
  dropDown: css({ maxWidth: '100%' }),
  roleListItem: css({
    display: 'grid',
    gridTemplateColumns: 'min-content auto',
    gridColumnGap: '4px',
    alignItems: 'center',
    color: tokens.colorTextDark
  }),
  adminSubtitle: css({
    color: tokens.colorTextLight,
    gridColumnStart: 2,
    maxWidth: '200px',
    whiteSpace: 'normal'
  }),
  rolesSummary: css({
    display: 'inline-block',
    overflowX: 'hidden',
    textOverflow: 'ellipsis'
  })
};

class SpaceRoleEditor extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    isDisabled: PropTypes.bool,
    options: PropTypes.arrayOf(SpaceRoleProp),
    value: PropTypes.arrayOf(PropTypes.string),
    className: PropTypes.string,
    buttonProps: PropTypes.shape({ className: PropTypes.string })
  };

  static defaultProps = {
    options: [],
    value: [],
    className: '',
    buttonProps: { className: '' }
  };

  state = {
    isOpen: false
  };

  setAdmin = () => {
    this.props.onChange([ADMIN_ROLE_ID]);
  };

  toggleRole = roleId => event => {
    event.stopPropagation();
    const isAdmin = roleId === ADMIN_ROLE_ID;
    const checked = this.props.value.includes(roleId);

    if (checked) {
      this.removeRole(roleId);
    } else {
      isAdmin ? this.setAdmin() : this.addRole(roleId);
    }
  };

  addRole(roleId) {
    const { value, onChange } = this.props;
    onChange(without(value, ADMIN_ROLE_ID).concat(roleId));
  }

  removeRole(roleId) {
    const { value, onChange } = this.props;
    onChange(value.filter(id => id !== roleId));
  }

  closeDropdown = () => {
    this.setState({ isOpen: false });
  };

  toggleDropdown = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    const { options, value, isDisabled, className, buttonProps } = this.props;
    const isAdmin = value.includes(ADMIN_ROLE_ID);

    const sortedOptions = [...options].sort(({ name: nameA }, { name: nameB }) =>
      nameA.localeCompare(nameB)
    );
    const selectedNames = [ADMIN_ROLE, ...sortedOptions]
      .filter(option => value.includes(option.sys.id))
      .map(role => role.name);

    let rolesSummary;
    if (selectedNames.length === 0) {
      rolesSummary = 'Select a space role';
    }
    if (selectedNames.length === 1) {
      rolesSummary = selectedNames[0];
    }
    if (selectedNames.length > 1) {
      rolesSummary = (
        <span className={css({ display: 'flex', alignItems: 'center' })}>
          <span className={styles.rolesSummary}>{selectedNames[0]}</span>&nbsp;and{' '}
          {selectedNames.length - 1} more
        </span>
      );
    }

    return (
      <Dropdown
        className={classNames(className, styles.dropDown)}
        isOpen={this.state.isOpen}
        onClose={this.closeDropdown}
        toggleElement={
          <Button
            testId="space-role-editor.button"
            disabled={isDisabled}
            className={`select-button${buttonProps.className ? ` ${buttonProps.className}` : ''}`}
            buttonType="naked"
            indicateDropdown
            onClick={this.toggleDropdown}>
            {rolesSummary}
          </Button>
        }>
        <div className={styles.roleList}>
          <DropdownList testId="space-role-editor.options">
            <DropdownListItem
              testId="space-role-editor.admin-option"
              onClick={this.toggleRole(ADMIN_ROLE_ID)}>
              <div className={styles.roleListItem}>
                <Checkbox labelText={ADMIN_ROLE.name} checked={isAdmin} id={ADMIN_ROLE_ID} />
                <div>{ADMIN_ROLE.name}</div>
                <div className={styles.adminSubtitle}>Can manage everything in the space</div>
              </div>
            </DropdownListItem>
          </DropdownList>
          {sortedOptions.length > 0 && (
            <DropdownList border="top">
              <DropdownListItem isTitle={true}>other roles</DropdownListItem>
              {sortedOptions.map(({ name, sys: { id } }) => (
                // Allow the whole list item to be clicked
                <DropdownListItem
                  key={id}
                  testId="space-role-editor.role-option"
                  onClick={this.toggleRole(id)}>
                  <div className={styles.roleListItem}>
                    {/* We don't use CheckboxField, as it emits double click events */}
                    {/* https://codesandbox.io/embed/cocky-wiles-r03rq */}
                    <Checkbox labelText={name} checked={value.includes(id)} id={id} />
                    <div>{name}</div>
                  </div>
                </DropdownListItem>
              ))}
            </DropdownList>
          )}
        </div>
      </Dropdown>
    );
  }
}

export default SpaceRoleEditor;
