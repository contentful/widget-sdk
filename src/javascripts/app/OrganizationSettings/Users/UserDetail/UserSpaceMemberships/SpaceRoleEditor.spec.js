import React from 'react';
import { mount } from 'enzyme';
import SpaceRoleEditor from './SpaceRoleEditor.es6';

describe('SpaceRoleEditor', () => {
  let changeCb;

  const roleOptions = [
    {
      name: 'Police officer',
      sys: { id: 'police' }
    },
    {
      name: 'Thief',
      sys: { id: 'thief' }
    }
  ];

  const render = props => {
    changeCb = jest.fn();
    const defaultProps = {
      options: roleOptions,
      value: [],
      onChange: changeCb
    };

    return mount(<SpaceRoleEditor {...Object.assign(defaultProps, props)} />);
  };

  beforeEach(() => {});

  it('renders the component', () => {
    expect(render()).toMatchSnapshot();
  });

  it('shows a list of options', () => {
    const component = render();
    component.find('button').simulate('click');
    expect(component).toMatchSnapshot();
  });

  it('removes the admin role if any other role is selected', () => {
    const component = render();
    component.props({ value: ['admin'] });
    component.find('button').simulate('click');
    component
      .find('input')
      .at(1)
      .simulate('change', { target: { checked: true } });
    expect(changeCb).toHaveBeenCalledWith(['police']);
  });

  it('removes any other role if admin is selected', () => {
    const component = render();
    component.props({ value: ['police', 'thief'] });
    component.find('button').simulate('click');
    component
      .find('input')
      .at(0)
      .simulate('change', { target: { checked: true } });
    expect(changeCb).toHaveBeenCalledWith(['admin']);
  });

  it('allows multiple roles', () => {
    const component = render({ value: ['police'] });
    component.find('button').simulate('click');
    component
      .find('input')
      .at(2)
      .simulate('change', { target: { checked: true } });
    expect(changeCb).toHaveBeenCalledWith(['police', 'thief']);
  });
});
