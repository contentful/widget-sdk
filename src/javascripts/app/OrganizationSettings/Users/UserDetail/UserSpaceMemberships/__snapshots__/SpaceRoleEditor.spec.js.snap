// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceRoleEditor renders the component 1`] = `
<MockedProvider
  services={
    Object {
      "access_control/SpaceMembershipRepository.es6": Object {
        "ADMIN_ROLE": Object {
          "name": "Admin",
          "sys": Object {
            "id": "admin",
          },
        },
        "ADMIN_ROLE_ID": "admin",
      },
    }
  }
>
  <ServicesConsumer
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "name": "Police officer",
          "sys": Object {
            "id": "police",
          },
        },
        Object {
          "name": "Thief",
          "sys": Object {
            "id": "thief",
          },
        },
      ]
    }
    value={Array []}
  >
    <SpaceRoleEditor
      $services={
        Object {
          "SpaceMembershipRepository": Object {
            "ADMIN_ROLE": Object {
              "name": "Admin",
              "sys": Object {
                "id": "admin",
              },
            },
            "ADMIN_ROLE_ID": "admin",
          },
        }
      }
      onChange={[MockFunction]}
      options={
        Array [
          Object {
            "name": "Police officer",
            "sys": Object {
              "id": "police",
            },
          },
          Object {
            "name": "Thief",
            "sys": Object {
              "id": "thief",
            },
          },
        ]
      }
      value={Array []}
    >
      <Dropdown
        isOpen={false}
        onClose={[Function]}
        position="bottom-left"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
        testId="cf-ui-dropdown"
        toggleElement={
          <Button
            buttonType="naked"
            disabled={false}
            extraClassNames="select-button"
            indicateDropdown={true}
            isFullWidth={false}
            loading={false}
            onClick={[Function]}
            testId="cf-ui-button"
          >
            Select a role
          </Button>
        }
      >
        <div
          className="Dropdown__Dropdown___1qyn8"
          data-test-id="cf-ui-dropdown"
          onClose={[Function]}
          position="bottom-left"
          style={
            Object {
              "maxWidth": "100%",
            }
          }
        >
          <Button
            buttonType="naked"
            disabled={false}
            extraClassNames="select-button"
            indicateDropdown={true}
            isFullWidth={false}
            loading={false}
            onClick={[Function]}
            testId="cf-ui-button"
          >
            <button
              className="Button__Button___1ZfFj a11y__focus-border--default___60AXp select-button Button__Button--naked___mB6LS"
              data-test-id="cf-ui-button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
            >
              <TabFocusTrap
                extraClassNames="Button__Button__inner-wrapper___3qrNC"
              >
                <span
                  className="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
                  tabIndex={-1}
                >
                  <span
                    className="Button__Button__label___3tcOj"
                  >
                    Select a role
                  </span>
                  <Icon
                    color="secondary"
                    icon="ArrowDown"
                    size="small"
                    testId="cf-ui-icon"
                  >
                    <ArrowDown
                      className="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
                      data-test-id="cf-ui-icon"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <svg
                        className="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
                        data-test-id="cf-ui-icon"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </ArrowDown>
                  </Icon>
                  <CSSTransition
                    classNames={
                      Object {
                        "enter": "Button__Button--spinner--enter___1qgg7",
                        "enterActive": "Button__Button--spinner-active___EEKjQ",
                        "exit": "Button__Button--spinner--exit___2RUI-",
                        "exitActive": "Button__Button--spinner-exit-active___3HXa7",
                      }
                    }
                    in={false}
                    mountOnEnter={true}
                    timeout={1000}
                    unmountOnExit={true}
                  >
                    <Transition
                      appear={false}
                      enter={true}
                      exit={true}
                      in={false}
                      mountOnEnter={true}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={1000}
                      unmountOnExit={true}
                    />
                  </CSSTransition>
                </span>
              </TabFocusTrap>
            </button>
          </Button>
        </div>
      </Dropdown>
    </SpaceRoleEditor>
  </ServicesConsumer>
</MockedProvider>
`;

exports[`SpaceRoleEditor shows a list of options 1`] = `
<MockedProvider
  services={
    Object {
      "access_control/SpaceMembershipRepository.es6": Object {
        "ADMIN_ROLE": Object {
          "name": "Admin",
          "sys": Object {
            "id": "admin",
          },
        },
        "ADMIN_ROLE_ID": "admin",
      },
    }
  }
>
  <ServicesConsumer
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "name": "Police officer",
          "sys": Object {
            "id": "police",
          },
        },
        Object {
          "name": "Thief",
          "sys": Object {
            "id": "thief",
          },
        },
      ]
    }
    value={Array []}
  >
    <SpaceRoleEditor
      $services={
        Object {
          "SpaceMembershipRepository": Object {
            "ADMIN_ROLE": Object {
              "name": "Admin",
              "sys": Object {
                "id": "admin",
              },
            },
            "ADMIN_ROLE_ID": "admin",
          },
        }
      }
      onChange={[MockFunction]}
      options={
        Array [
          Object {
            "name": "Police officer",
            "sys": Object {
              "id": "police",
            },
          },
          Object {
            "name": "Thief",
            "sys": Object {
              "id": "thief",
            },
          },
        ]
      }
      value={Array []}
    >
      <Dropdown
        isOpen={true}
        onClose={[Function]}
        position="bottom-left"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
        testId="cf-ui-dropdown"
        toggleElement={
          <Button
            buttonType="naked"
            disabled={false}
            extraClassNames="select-button"
            indicateDropdown={true}
            isFullWidth={false}
            loading={false}
            onClick={[Function]}
            testId="cf-ui-button"
          >
            Select a role
          </Button>
        }
      >
        <div
          className="Dropdown__Dropdown___1qyn8"
          data-test-id="cf-ui-dropdown"
          onClose={[Function]}
          position="bottom-left"
          style={
            Object {
              "maxWidth": "100%",
            }
          }
        >
          <Button
            buttonType="naked"
            disabled={false}
            extraClassNames="select-button"
            indicateDropdown={true}
            isFullWidth={false}
            loading={false}
            onClick={[Function]}
            testId="cf-ui-button"
          >
            <button
              className="Button__Button___1ZfFj a11y__focus-border--default___60AXp select-button Button__Button--naked___mB6LS"
              data-test-id="cf-ui-button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
            >
              <TabFocusTrap
                extraClassNames="Button__Button__inner-wrapper___3qrNC"
              >
                <span
                  className="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
                  tabIndex={-1}
                >
                  <span
                    className="Button__Button__label___3tcOj"
                  >
                    Select a role
                  </span>
                  <Icon
                    color="secondary"
                    icon="ArrowDown"
                    size="small"
                    testId="cf-ui-icon"
                  >
                    <ArrowDown
                      className="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
                      data-test-id="cf-ui-icon"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <svg
                        className="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
                        data-test-id="cf-ui-icon"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </ArrowDown>
                  </Icon>
                  <CSSTransition
                    classNames={
                      Object {
                        "enter": "Button__Button--spinner--enter___1qgg7",
                        "enterActive": "Button__Button--spinner-active___EEKjQ",
                        "exit": "Button__Button--spinner--exit___2RUI-",
                        "exitActive": "Button__Button--spinner-exit-active___3HXa7",
                      }
                    }
                    in={false}
                    mountOnEnter={true}
                    timeout={1000}
                    unmountOnExit={true}
                  >
                    <Transition
                      appear={false}
                      enter={true}
                      exit={true}
                      in={false}
                      mountOnEnter={true}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={1000}
                      unmountOnExit={true}
                    />
                  </CSSTransition>
                </span>
              </TabFocusTrap>
            </button>
          </Button>
          <div
            className="Dropdown__Dropdown__container___1wpIB Dropdown__Dropdown__container-position--bottom-left___jqo6Y"
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <InViewport
              offset={0}
              onOverflowBottom={[Function]}
              onOverflowLeft={[Function]}
              onOverflowRight={[Function]}
              onOverflowTop={[Function]}
              testId="cf-ui-in-viewport"
            >
              <div
                className="InViewport__InViewport___2o6g4"
                data-test-id="cf-ui-in-viewport"
                offset={0}
              >
                <DropdownList
                  maxHeight={300}
                  testId="cf-ui-dropdown-list"
                >
                  <ul
                    className="DropdownList__DropdownList___2EMLM"
                    data-test-id="cf-ui-dropdown-list"
                    style={
                      Object {
                        "maxHeight": 300,
                        "overflowY": "auto",
                      }
                    }
                  >
                    <DropdownListItem
                      isActive={false}
                      isDisabled={false}
                      isTitle={false}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      testId="cf-ui-dropdown-list-item"
                    >
                      <li
                        className="DropdownListItem__DropdownListItem___LOUnP"
                        data-test-id="cf-ui-dropdown-list-item"
                      >
                        <CheckboxField
                          checked={false}
                          id="admin"
                          inputType="checkbox"
                          labelIsLight={false}
                          labelText="Admin"
                          light={true}
                          onChange={[Function]}
                          testId="cf-ui-checkbox-field"
                        >
                          <ControlledInputField
                            checked={false}
                            id="admin"
                            inputType="checkbox"
                            labelIsLight={false}
                            labelText="Admin"
                            light={true}
                            onChange={[Function]}
                            testId="cf-ui-checkbox-field"
                          >
                            <div
                              className="ControlledInputField__ControlledInputField___2uIG9"
                              data-test-id="cf-ui-checkbox-field"
                              light={true}
                            >
                              <ControlledInput
                                checked={false}
                                disabled={false}
                                extraClassNames="ControlledInputField__ControlledInputField__input___3OMYB"
                                id="admin"
                                labelText="Admin"
                                onChange={[Function]}
                                required={false}
                                testId="cf-ui-controlled-input"
                                type="checkbox"
                              >
                                <input
                                  aria-label="Admin"
                                  checked={false}
                                  className="ControlledInput__ControlledInput___2XK3j ControlledInputField__ControlledInputField__input___3OMYB"
                                  data-test-id="cf-ui-controlled-input"
                                  disabled={false}
                                  id="admin"
                                  onChange={[Function]}
                                  required={false}
                                  type="checkbox"
                                />
                              </ControlledInput>
                              <div>
                                <FormLabel
                                  extraClassNames="ControlledInputField__ControlledInputField__label___a9J52"
                                  htmlFor="admin"
                                  required={false}
                                  requiredText="required"
                                  testId="cf-ui-form-label"
                                >
                                  <label
                                    className="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
                                    data-test-id="cf-ui-form-label"
                                    htmlFor="admin"
                                  >
                                    Admin
                                  </label>
                                </FormLabel>
                              </div>
                            </div>
                          </ControlledInputField>
                        </CheckboxField>
                      </li>
                    </DropdownListItem>
                    <DropdownListItem
                      isActive={false}
                      isDisabled={false}
                      isTitle={false}
                      key="police"
                      onEnter={[Function]}
                      onLeave={[Function]}
                      testId="cf-ui-dropdown-list-item"
                    >
                      <li
                        className="DropdownListItem__DropdownListItem___LOUnP"
                        data-test-id="cf-ui-dropdown-list-item"
                      >
                        <CheckboxField
                          checked={false}
                          id="police"
                          inputType="checkbox"
                          labelIsLight={false}
                          labelText="Police officer"
                          light={true}
                          onChange={[Function]}
                          testId="cf-ui-checkbox-field"
                          value="police"
                        >
                          <ControlledInputField
                            checked={false}
                            id="police"
                            inputType="checkbox"
                            labelIsLight={false}
                            labelText="Police officer"
                            light={true}
                            onChange={[Function]}
                            testId="cf-ui-checkbox-field"
                            value="police"
                          >
                            <div
                              className="ControlledInputField__ControlledInputField___2uIG9"
                              data-test-id="cf-ui-checkbox-field"
                              light={true}
                            >
                              <ControlledInput
                                checked={false}
                                disabled={false}
                                extraClassNames="ControlledInputField__ControlledInputField__input___3OMYB"
                                id="police"
                                labelText="Police officer"
                                onChange={[Function]}
                                required={false}
                                testId="cf-ui-controlled-input"
                                type="checkbox"
                                value="police"
                              >
                                <input
                                  aria-label="Police officer"
                                  checked={false}
                                  className="ControlledInput__ControlledInput___2XK3j ControlledInputField__ControlledInputField__input___3OMYB"
                                  data-test-id="cf-ui-controlled-input"
                                  disabled={false}
                                  id="police"
                                  onChange={[Function]}
                                  required={false}
                                  type="checkbox"
                                  value="police"
                                />
                              </ControlledInput>
                              <div>
                                <FormLabel
                                  extraClassNames="ControlledInputField__ControlledInputField__label___a9J52"
                                  htmlFor="police"
                                  required={false}
                                  requiredText="required"
                                  testId="cf-ui-form-label"
                                >
                                  <label
                                    className="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
                                    data-test-id="cf-ui-form-label"
                                    htmlFor="police"
                                  >
                                    Police officer
                                  </label>
                                </FormLabel>
                              </div>
                            </div>
                          </ControlledInputField>
                        </CheckboxField>
                      </li>
                    </DropdownListItem>
                    <DropdownListItem
                      isActive={false}
                      isDisabled={false}
                      isTitle={false}
                      key="thief"
                      onEnter={[Function]}
                      onLeave={[Function]}
                      testId="cf-ui-dropdown-list-item"
                    >
                      <li
                        className="DropdownListItem__DropdownListItem___LOUnP"
                        data-test-id="cf-ui-dropdown-list-item"
                      >
                        <CheckboxField
                          checked={false}
                          id="thief"
                          inputType="checkbox"
                          labelIsLight={false}
                          labelText="Thief"
                          light={true}
                          onChange={[Function]}
                          testId="cf-ui-checkbox-field"
                          value="thief"
                        >
                          <ControlledInputField
                            checked={false}
                            id="thief"
                            inputType="checkbox"
                            labelIsLight={false}
                            labelText="Thief"
                            light={true}
                            onChange={[Function]}
                            testId="cf-ui-checkbox-field"
                            value="thief"
                          >
                            <div
                              className="ControlledInputField__ControlledInputField___2uIG9"
                              data-test-id="cf-ui-checkbox-field"
                              light={true}
                            >
                              <ControlledInput
                                checked={false}
                                disabled={false}
                                extraClassNames="ControlledInputField__ControlledInputField__input___3OMYB"
                                id="thief"
                                labelText="Thief"
                                onChange={[Function]}
                                required={false}
                                testId="cf-ui-controlled-input"
                                type="checkbox"
                                value="thief"
                              >
                                <input
                                  aria-label="Thief"
                                  checked={false}
                                  className="ControlledInput__ControlledInput___2XK3j ControlledInputField__ControlledInputField__input___3OMYB"
                                  data-test-id="cf-ui-controlled-input"
                                  disabled={false}
                                  id="thief"
                                  onChange={[Function]}
                                  required={false}
                                  type="checkbox"
                                  value="thief"
                                />
                              </ControlledInput>
                              <div>
                                <FormLabel
                                  extraClassNames="ControlledInputField__ControlledInputField__label___a9J52"
                                  htmlFor="thief"
                                  required={false}
                                  requiredText="required"
                                  testId="cf-ui-form-label"
                                >
                                  <label
                                    className="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
                                    data-test-id="cf-ui-form-label"
                                    htmlFor="thief"
                                  >
                                    Thief
                                  </label>
                                </FormLabel>
                              </div>
                            </div>
                          </ControlledInputField>
                        </CheckboxField>
                      </li>
                    </DropdownListItem>
                  </ul>
                </DropdownList>
              </div>
            </InViewport>
          </div>
        </div>
      </Dropdown>
    </SpaceRoleEditor>
  </ServicesConsumer>
</MockedProvider>
`;
